client/relay-chain-rpc-interface/src/tokio_platform.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
```

Comment:
I just wanted to mention that the code that you're adapting is a bit of a "shitshow"

Since it's an implementation detail within smoldot I didn't pay much attention to it, but it's definitely meant to be rewritten at some point